components/expenses/PayExpenseWithStripe.tsx

```diff
@@ -0,0 +1,255 @@
+    [selectedPaymentMethodId, savedPaymentMethods, paymentElementData],
+  );
+
+  const { onSuccess } = props;
+  const onPayClick = React.useCallback(async () => {
+    const returnUrl = new URL(
+      `${window.location.origin}/${props.expense.account.slug}/expenses/${props.expense.legacyId}`,
+    );
+
+    const selectedPaymentMethod = savedPaymentMethods.find(pm => pm.id === selectedPaymentMethodId);
+    try {
+      setIsPayButtonLoading(true);
+      await confirmPayment(
+        stripe,
+        paymentIntent?.client_secret,
+        {
+          returnUrl: returnUrl.href,
+          elements: paymentElementData?.stripeData?.elements,
+          type:
+            selectedPaymentMethodId === 'new-payment-method'
+              ? paymentElementData?.paymentMethod?.type
+              : selectedPaymentMethod.type,
+          paymentMethodId:
+            selectedPaymentMethodId !== 'new-payment-method'
+              ? selectedPaymentMethod?.data?.stripePaymentMethodId
+              : undefined,
+        },
+        {
+          redirect: 'if_required',
+        },
+      );
+      onSuccess();
+    } catch (err) {
+      toast({ variant: 'error', message: err.message });
+    } finally {
+      setIsPayButtonLoading(false);
+    }
+  }, [
+    stripe,
+    paymentIntent,
+    paymentElementData,
+    selectedPaymentMethodId,
+    props.expense.legacyId,
+    props.expense.account.slug,
+    savedPaymentMethods,
+    onSuccess,
+  ]);
+
+  if (error) {
+    return <MessageBoxGraphqlError error={error} />;
+  } else if (loading) {
+    return <Skeleton className="h-6 w-full" />;
+  }
+
+  const stripeForm = (
+    <PayWithStripeForm
+      hasSaveCheckBox={false}
+      defaultIsSaved={false}
+      onChange={e => setPaymentElementData(e.stepPayment)}
+      bilingDetails={{
+        name: '',
+        email: '',
+      }}
+      paymentIntentId={paymentIntent.id}
+      paymentIntentClientSecret={paymentIntent.client_secret}
+    />
+  );
+
+  return (
+    <Elements options={options} stripe={stripe}>
+      {savedPaymentMethods?.length ? (
+        <RadioGroup id="paymentMethod" value={selectedPaymentMethodId} onValueChange={setSelectedPaymentMethodId}>
+          {savedPaymentMethods.map(pm => (
+            <RadioGroupCard key={pm.id} value={pm.id}>
+              <div>{getPaymentMethodIcon(pm)}</div>
+              <div>
+                <div className="text-sm">{getPaymentMethodName(pm as any)}</div>
+                <div className="text-xs text-muted-foreground">{getPaymentMethodMetadata(pm)}</div>
+              </div>
+            </RadioGroupCard>
+          ))}
+
+          <RadioGroupCard
+            value="new-payment-method"
+            showSubcontent={selectedPaymentMethodId === 'new-payment-method'}
+            subContent={stripeForm}
+          >
+            <FormattedMessage
+              defaultMessage="New: {methods}"
+              id="F9QkrV"
```

Comment:
This message will be difficult to translate as-is, I suggest adding a bit of context:

```suggestion
              defaultMessage="New: {methods}"
              id="paymentMethods.new"
```